dandiapi/api/services/embargo/__init__.py

```diff
@@ -33,28 +38,65 @@ def _delete_asset_blob_tags(client: S3Client, blob: str):
 
 # NOTE: In testing this took ~2 minutes for 100,000 files
 def _remove_dandiset_asset_blob_embargo_tags(dandiset: Dandiset):
-    # First we need to generate a CSV manifest containing all asset blobs that need to be untaged
-    embargoed_asset_blobs = AssetBlob.objects.filter(
-        embargoed=True, assets__versions__dandiset=dandiset
-    ).values_list('blob', flat=True)
+    embargoed_asset_blobs = list(
+        AssetBlob.objects.filter(embargoed=True, assets__versions__dandiset=dandiset).values_list(
+            'blob', flat=True
+        )
+    )
 
     client = get_boto_client(config=Config(max_pool_connections=100))
     with ThreadPoolExecutor(max_workers=100) as e:
-        for blob in embargoed_asset_blobs:
+        futures = [
             e.submit(_delete_asset_blob_tags, client=client, blob=blob)
+            for blob in embargoed_asset_blobs
+        ]
+
+    # Check if any failed and raise exception if so
+    failed = [
+        blob for i, blob in enumerate(embargoed_asset_blobs) if futures[i].exception() is not None
+    ]
```

Comment:
According to the [`Future` docs](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.exception) the `.exception()` method will wait forever for the job to complete without a `timeout` parameter. If something goes wrong with `_delete_asset_blob_tags`, what do we do to recover?

Reply:
The `ThreadPoolExecutor` context manager only exits once all futures have completed, so by the time this line is reached, we can be sure all futures have completed, successfully or otherwise. `.exception()` will then return (not raise) an exception if any were raised in `_delete_asset_blob_tags`. Otherwise, it returns `None`.

Reply:
We discussed offline and decided that this specific problem isn't much of an issue, but that more care should be taken around timeouts. This will be fully addressed in follow-on work.